src/bridges/set/IssuanceBridge.sol

```diff
@@ -3,40 +3,62 @@
 pragma solidity >=0.6.10 <=0.8.10;
 pragma experimental ABIEncoderV2;
 
-import { SafeMath } from "@openzeppelin/contracts/utils/math/SafeMath.sol";
-import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
 
-import { AztecTypes } from "../../aztec/AztecTypes.sol";
+import {AztecTypes} from '../../aztec/AztecTypes.sol';
 
-import { IDefiBridge } from "../../interfaces/IDefiBridge.sol";
-import { ISetToken } from "./interfaces/ISetToken.sol";
-import { IController } from "./interfaces/IController.sol";
-import { IExchangeIssuance } from "./interfaces/IExchangeIssuance.sol";
-import { IRollupProcessor } from "../../interfaces/IRollupProcessor.sol";
+import {IDefiBridge} from '../../interfaces/IDefiBridge.sol';
+import {ISetToken} from './interfaces/ISetToken.sol';
+import {IController} from './interfaces/IController.sol';
+import {IExchangeIssuance} from './interfaces/IExchangeIssuance.sol';
+import {IRollupProcessor} from '../../interfaces/IRollupProcessor.sol';
 
 contract IssuanceBridge is IDefiBridge {
-    using SafeMath for uint256;
+    IExchangeIssuance public constant EXCHANGE_ISSUANCE = IExchangeIssuance(0xc8C85A3b4d03FB3451e7248Ff94F780c92F884fD);
+    IController public constant SET_CONTROLLER = IController(0xa4c8d221d8BB851f83aadd0223a8900A6921A349);
 
     address public immutable rollupProcessor;
 
-    IExchangeIssuance exchangeIssuance;
-    IController setController; // used to check if address is SetToken
+    error ApproveFailed(address token);
+    error InvalidCaller();
+    error ZeroInputValue();
+    error IncorrectInput();
+    error AsyncModeDisabled();
 
-    constructor(
-        address _rollupProcessor,
-        address _exchangeIssuance,
-        address _setController
-    ) public {
+    /**
+     * @notice Sets the addresses of RollupProcessor
+     * @param _rollupProcessor Address of the RollupProcessor.sol
+     */
+    constructor(address _rollupProcessor) public {
         rollupProcessor = _rollupProcessor;
-        exchangeIssuance = IExchangeIssuance(_exchangeIssuance);
-        setController = IController(_setController);
+    }
+
+    // Empty receive function for the contract to be able to receive ETH
+    receive() external payable {}
+
+    /**
+     * @notice Sets all the important approvals.
+     * @dev This bridge never holds any ERC20s after or before an invocation of any of its functions.
+     * For this reason the following is not a security risk and makes the convert() function more gas efficient.
+     */
+    function approveTokens(address[] calldata tokens) external {
+        uint256 tokensLength = tokens.length;
+        for (uint256 i; i < tokensLength; ) {
+            if (!IERC20(tokens[i]).approve(rollupProcessor, type(uint256).max)) revert ApproveFailed(tokens[i]);
```

Comment:
Maybe SafeApprove instead? Would be annoying to have specific assets not useable

Reply:
I think that here it might not be necessary because of what Set protocol is focused on. If I am not mistaken Tether is the only important token which requires this and I don't think it makes sense for Set to create a basket containing Tether (if you want to hold multiple stables you can just hold curve pool token).